visor/config/run_config.go

```diff
@@ -18,28 +18,134 @@ import (
 )
 
+/*
+description: Allows to configure binary and it's arguments.
+example:
+
+	type: toml
+	value: |
+		path = "/path/binary"
+		args = ["--arg1", "val1", "--arg2"]
+*/
 type BinaryConfig struct {
-	Path string   `toml:"path"`
+	/*
+		description: Path to the binary.
+		note: |
+			Both absolute or relative path can be used.
+			Relative path is relative to a parent folder of this config file.
+	*/
+	Path string `toml:"path"`
+	/*
+		description: Arguments that will be applied to the binary.
+		note: |
+			Each element the list represents one space seperated argument.
+	*/
 	Args []string `toml:"args"`
 }
 
+/*
+description: Allows to configure connection to core node exposed UNIX socket RPC API.
+example:
+
+	type: toml
+	value: |
+		[vega.rpc]
+			socketPath = "/path/socket.sock"
+			httpPath = "/rpc"
+*/
 type RPCConfig struct {
+	/*
+		description: Path of the mounted socket.
+		note: This path can be configured in Vega core node configuration.
+	*/
 	SocketPath string `toml:"socketPath"`
-	HTTPPath   string `toml:"httpPath"`
+	/*
+		description: HTTP path of the socket path.
+		note: This path can be configured in Vega core node configuration.
+	*/
+	HTTPPath string `toml:"httpPath"`
 }
 
+/*
+description: Allows to configure Vega binary and it's arguments.
+example:
+
+	type: toml
+	value: |
+		[vega]
+			[vega.binary]
+				path = "/path/vega-binary"
+				args = ["--arg1", "val1", "--arg2"]
+			[vega.rpc]
+				socketPath = "/path/socket.sock"
+				httpPath = "/rpc"
+*/
 type VegaConfig struct {
+	/*
+		description: Configuration of Vega binary to be run.
+		example:
+			type: toml
+			value: |
+				[vega.binary]
+					path = "/path/vega-binary"
+					args = ["--arg1", "val1", "--arg2"]
+	*/
 	Binary BinaryConfig `toml:"binary"`
-	RCP    RPCConfig    `toml:"rpc"`
+
+	/*
+		description: |
+			Visor communicates with the core node via RPC API that runs over UNIX socket.
+			This parameter allows to configure the UNIX socket to match the core node configuration.
+		example:
+			type: toml
+			value: |
+				[vega.binary]
+					path = "/path/vega-binary"
+					args = ["--arg1", "val1", "--arg2"]
+	*/
+	RCP RPCConfig `toml:"rpc"`
 }
 
+/*
+description: Allows to configure Data node binary and it's arguments.
+example:
+
+	type: toml
+	value: |
+		[data_node]
+			[data_node.binary]
+				path = "/path/data-node-binary"
+				args = ["--arg1", "val1", "--arg2"]
+*/
 type DataNodeConfig struct {
 	Binary BinaryConfig `toml:"binary"`
 }
 
+/*
+description: Root of the config file
+example:
+
+	type: toml
+	value: |
+		name = "v1.65.0"
+
+		[vega]
+			[vega.binary]
+				path = "/path/vega-binary"
+				args = ["--arg1", "val1", "--arg2"]
+			[vega.rpc]
+				socketPath = "/path/socket.sock"
+				httpPath = "/rpc"
+*/
 type RunConfig struct {
-	Name     string          `toml:"name"`
-	Vega     VegaConfig      `toml:"vega"`
+	/*
+		description: Name of the upgrade.
+		note: It is recommended to use an upgrade version as a name.
```

Comment:
```suggestion
		note: It is recommended to use the upgrade version as the name (for example: 0.50.0).
```
@karlem is the example I added correct?